### PR TITLE
[blog] Add Scaffolding blog posts.

### LINKED ISSUE #2272: Write a blog about Scaffolding: Simple 'tutorial' post with example.
Write a blog about Scaffolding: Simple 'tutorial' post with example.

Needs to be handed over by May 15.

### PR DESCRIPTION
**Pull request by fnichol**
Closes #2272

gif-keyboard-4934600215194553302

### CODE REVIEW

**Comment by thesentinels** [comment]
Thanks for the pull request! Here is what will happen next:
 1. Your PR will be reviewed by the maintainers
 2. If everything looks good, one of them will `approve` it, and your PR will be merged.

Thank you for contributing!

File: www/source/blog/2017-05-23-Scaffolding-App-From-Scratch.html.md [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+wget -q -O - http://localhost:8000
+~~~
+
+I'm hoping you're looking at your fabulously generated HTML, just as I am:
+
+~~~
+<!DOCTYPE html><html><head><title>Express</title><link rel="stylesheet" href="/stylesheets/style.css"></head><body><h1>Express</h1><p>Welcome to Express</p></body></html>
+~~~
+
+Finally, how about performing a next-level Habitat trick like changing your app's listen port using a versioned runtime configuration injected into Habitat's gossip ring? The Node Scaffolding has provided us with a called `app.port` which we can change, so let's set it to `3000` instead:
```
Comment (mwrock) [inline www/source/blog/2017-05-23-Scaffolding-App-From-Scratch.html.md]: With a configuration export called

File: www/source/blog/2017-05-23-Scaffolding-App-From-Scratch.html.md [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+~~~sh
+pkg_origin=fnichol
+pkg_name=expresso
+pkg_version=0.1.0
+pkg_scaffolding=core/scaffolding-node
+~~~
+
+Notice that I used my Habitat origin of `fnichol` (that's the origin and key I generated when I ran `hab setup` earlier on). The last line is where the good stuff is--this tells the Habitat build system to use the `core/scaffolding-node` Scaffolding package to build your app.
+
+I'm sure your as antsy as I am to commit this to version control--let's do that now:
```
Comment (mwrock) [inline www/source/blog/2017-05-23-Scaffolding-App-From-Scratch.html.md]: Sure you are

File: www/source/blog/2017-05-23-Scaffolding.html.md [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+
+We're launching this feature with two Scaffolding packages: one for [**Ruby**](https://github.com/habitat-sh/core-plans/tree/master/scaffolding-ruby)-based apps and one for [**Node.js**](https://github.com/habitat-sh/core-plans/tree/master/scaffolding-node)-based apps. In the near future, we plan on adding support for Go, Python, and JVM based apps but we believe we have a good enough start to share this build pattern more widely.
+
+## Scaffolding's Assumptions
+
+First, this feature is obsessively focused on the application developer's experience. As a result what we concerned with are developers building their own applications. Habitat is developing a good track record for building **other** third-party software (for example: [PostgreSQL](https://app.habitat.sh/#/pkgs/core/postgresql), [Bash](https://app.habitat.sh/#/pkgs/core/bash), [Rust](https://app.habitat.sh/#/pkgs/core/rust), etc.), but as a developer, I want Habitat to be great at building **my** software. We expect and assume that you want your Habitat Plan to live alongside your codebase, in the same version control repository, etc.
```
Comment (mwrock) [inline www/source/blog/2017-05-23-Scaffolding.html.md]: As a result, what we are concerned with

File: www/source/blog/2017-05-23-Scaffolding.html.md [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+   habirails: Detected Rails 5 app type
+~~~
+
+The Ruby Scaffolding understands some specific Ruby web frameworks such as Ruby on Rails and Rack. In this case it has detected a Rails 5.x application and can use that knowledge later on.
+
+~~~
+   habirails: No Ruby version detected in Plan or Gemfile.lock, using default 'core/ruby'
+~~~
+
+There are canonical locations where Ruby developers select a specific version Ruby, one of them uses the `ruby` keyword in a `Gemfile`. The Ruby Scaffolding loads the project's `Gemfile` and `Gemfile.lock` (using Ruby itself and calls the Bundler codebase as a library) so that it can correctly parse this information. In this case no version was specified in the `Gemfile` or in the `plan.sh`, so a default Habitat package of `core/ruby` was chosen.
```
Comment (mwrock) [inline www/source/blog/2017-05-23-Scaffolding.html.md]: Version of ruby

**Comment by mwrock** [comment]
Looks really really good! Added a few corrections.

**Comment by fnichol** [comment]
@mwrock Thanks, added your fixes!

**Comment by mwrock** [comment]
@thesentinels approve

**Comment by thesentinels** [comment]
:metal: I am testing your branch against master before merging it. We do this to ensure that the master branch is never failing tests.

**Comment by thesentinels** [comment]
:neckbeard: Travis CI has [started testing this PR](https://travis-ci.org/habitat-sh/habitat/builds/234553604).

**Comment by thesentinels** [comment]
:sparkling_heart: Travis CI [reports this PR passed](https://travis-ci.org/habitat-sh/habitat/builds/234553604).

It always makes me feel nice when humans approve of one anothers work. I'm merging this PR now.

I just want you and the contributor to answer me one question:

gif-keyboard-3280869874741411265

**Pull request merged**